corehq/util/datadog/__init__.py

```diff
@@ -0,0 +1,37 @@
+from django.conf import settings
+import logging
+
+logger = logging.getLogger(__name__)
+
+try:
+    from datadog.dogstatsd.base import DogStatsd
+    use_real_statsd = True
+except ImportError:
+    use_real_statsd = False
+
+
+class Callable(object):
+    def __init__(self, name, ret=None):
+        self.name = name
+        self.ret = ret
+
+    def __call__(self, *args, **kwargs):
+        logger.debug("mock call: statsd.%s(*%s, **%s)", self.name, args, kwargs)
+        return self.ret
+
+
+class MockStatsd(object):
+    def __getattribute__(self, item):
+        if item == 'timed':
+            def no_op_decorator(fn):
+                return fn
+            return Callable(item, ret=no_op_decorator)
+        return Callable(item)
```

Comment:
There are other functions that can also be called and I didn't want to have to implement them all (lazy)